src/diffusers/models/transformers/transformer_wan_animate.py

```diff
@@ -0,0 +1,889 @@
+            out_dim=motion_encoder_dim,
+            channels=motion_encoder_channel_sizes,
+        )
+
+        # Face encoder
+        self.face_encoder = WanAnimateFaceEncoder(
+            in_dim=motion_encoder_dim,
+            out_dim=inner_dim,
+            hidden_dim=face_encoder_hidden_dim,
+            num_heads=face_encoder_num_heads,
+        )
+
+        # 3. Transformer blocks
+        self.blocks = nn.ModuleList(
+            [
+                WanTransformerBlock(
+                    dim=inner_dim,
+                    ffn_dim=ffn_dim,
+                    num_heads=num_attention_heads,
+                    qk_norm=qk_norm,
+                    cross_attn_norm=cross_attn_norm,
+                    eps=eps,
+                    added_kv_proj_dim=added_kv_proj_dim,
+                )
+                for _ in range(num_layers)
+            ]
+        )
+
+        self.face_adapter = nn.ModuleList(
+            [
+                WanAnimateFaceBlockCrossAttention(
+                    dim=inner_dim,
+                    heads=num_attention_heads,
+                    dim_head=inner_dim // num_attention_heads,
+                    eps=eps,
+                    cross_attention_dim_head=inner_dim // num_attention_heads,
+                    processor=WanAnimateFaceBlockAttnProcessor(),
+                )
+                for _ in range(num_layers // inject_face_latents_blocks)
+            ]
+        )
+
+        # 4. Output norm & projection
+        self.norm_out = FP32LayerNorm(inner_dim, eps, elementwise_affine=False)
+        self.proj_out = nn.Linear(inner_dim, out_channels * math.prod(patch_size))
+        self.scale_shift_table = nn.Parameter(torch.randn(1, 2, inner_dim) / inner_dim**0.5)
+
+        self.gradient_checkpointing = False
+
+    def motion_batch_encode(
+        self, face_pixel_values: torch.Tensor, batch_size: int = 8, upcast_to_fp32: bool = True
+    ) -> torch.Tensor:
+        if batch_size >= face_pixel_values.shape[0]:
+            motion_vec = self.motion_encoder(face_pixel_values, upcast_to_fp32)
+        else:
+            motion_vec_batches = []
+            for i in range(math.ceil(face_pixel_values.shape[0] / batch_size)):
+                face_pixel_values_batch = face_pixel_values[i * batch_size : (i + 1) * batch_size]
+                motion_vec_batch = self.motion_encoder(face_pixel_values_batch, upcast_to_fp32)
+                motion_vec_batches.append(motion_vec_batch)
+            motion_vec = torch.cat(motion_vec_batches)
+
+        return motion_vec
+
+    def forward(
+        self,
+        hidden_states: torch.Tensor,
+        timestep: torch.LongTensor,
+        encoder_hidden_states: torch.Tensor,
+        encoder_hidden_states_image: Optional[torch.Tensor] = None,
+        pose_hidden_states: Optional[torch.Tensor] = None,
+        face_pixel_values: Optional[torch.Tensor] = None,
+        motion_encode_batch_size: int = 8,
+        upcast_to_fp32: bool = True,
```

Comment:
if we need fp32 to have good result, let's just use fp32, remove this argument - not currently used in the pipeline anyway